server/manager/src/main/java/org/apache/accumulo/manager/state/SetGoalState.java

```diff
@@ -45,12 +44,12 @@ public static void main(String[] args) throws Exception {
     try {
       var siteConfig = SiteConfiguration.auto();
       SecurityUtil.serverLogin(siteConfig);
-      var context = new ServerContext(siteConfig);
-      RenameMasterDirInZK.renameMasterDirInZK(context);
-      context.waitForZookeeperAndHdfs();
-      context.getZooReaderWriter().putPersistentData(
-          context.getZooKeeperRoot() + Constants.ZMANAGER_GOAL_STATE, args[0].getBytes(UTF_8),
-          NodeExistsPolicy.OVERWRITE);
+      try (var context = new ServerContext(siteConfig)) {
```

Comment:
Is it safe to have `close()` called on ServerContext here?  The try-with-resources will be a slight change in behavior.

Reply:
Yes, it's safe in this `main` method for `SetGoalState`. We're exiting immediately after anyway. The only thing my change did was make sure the resource was closed so it shut down any internal thread pools before the process exits. This got rid of a warning after removing the other line. I'm not sure why my IDE didn't see the warning before, but it didn't. It only noticed the unclosed resource after I deleted the other line.

Reply:
Closing a closable resource is the normal case. There's nothing to comment on. Being unable to close it would be unusual... but in the cases where we can't close it... it's because it's still being used. Also not unusual.

I'm not sure why somebody would think that this *wasn't* safe.

Reply:
Basically - calling close on the server context in the manager seem that it could be unintentional / unsafe if the context was being used elsewhere - it would take someone looking at the code additional time to figure that out and a simple comment might save someone same time.

Reply:
The entire contents of the file could fit on a small Post-It note (if you omit the import statements and license header). I don't think it will take anybody any substantial amount of time to realize that context is not being used anywhere else. And there's no special or unique situation to call out. So, I'd struggle to even know what I could say that would be helpful in a comment.